public/pages/workflow_detail/utils/workflow_to_template_utils.ts

```diff
@@ -59,12 +70,66 @@ function configToProvisionTemplateFlow(config: WorkflowConfig): TemplateFlow {
   };
 }
 
-// General fn to process all processor configs. Generate a final
-// ingest pipeline containing all of the processors, maintaining order
-function processorConfigsToTemplateNodes(
-  processorConfigs: IProcessorConfig[]
+function ingestConfigToTemplateNodes(
+  ingestConfig: IngestConfig
+): TemplateNode[] {
+  const ingestPipelineName = generateId('ingest_pipeline');
+  const ingestProcessors = processorConfigsToTemplateProcessors(
+    ingestConfig.enrich.processors
+  );
+
+  return ingestProcessors.length > 0
+    ? [
+        {
+          id: ingestPipelineName,
+          type: WORKFLOW_STEP_TYPE.CREATE_INGEST_PIPELINE_STEP_TYPE,
+          user_inputs: {
+            pipeline_id: ingestPipelineName,
+            configurations: {
+              description: 'An ingest pipeline',
+              processors: ingestProcessors,
+            },
+          },
+        } as CreateIngestPipelineNode,
+      ]
+    : [];
+}
+
+function searchConfigToTemplateNodes(
+  searchConfig: SearchConfig
 ): TemplateNode[] {
-  const processorsList = [] as IngestProcessor[];
+  const searchPipelineName = generateId('search_pipeline');
+  const searchRequestProcessors = processorConfigsToTemplateProcessors(
+    searchConfig.enrichRequest.processors
+  );
+  const searchResponseProcessors = processorConfigsToTemplateProcessors(
+    searchConfig.enrichResponse.processors
+  );
+
+  return searchRequestProcessors.length > 0 ||
+    searchResponseProcessors.length > 0
+    ? [
+        {
+          id: searchPipelineName,
+          type: WORKFLOW_STEP_TYPE.CREATE_SEARCH_PIPELINE_STEP_TYPE,
+          user_inputs: {
+            pipeline_id: searchPipelineName,
+            configurations: {
+              request_processors: searchRequestProcessors,
+              response_processors: searchResponseProcessors,
+            },
+          },
+        } as CreateSearchPipelineNode,
+      ]
+    : [];
```

Comment:
```suggestion
  const hasProcessors = searchRequestProcessors.length > 0 || searchResponseProcessors.length > 0;

return hasProcessors
  ? [
      {
        id: searchPipelineName,
        type: WORKFLOW_STEP_TYPE.CREATE_SEARCH_PIPELINE_STEP_TYPE,
        user_inputs: {
          pipeline_id: searchPipelineName,
          configurations: {
            request_processors: searchRequestProcessors,
            response_processors: searchResponseProcessors,
          },
        },
      } as CreateSearchPipelineNode,
    ]
  : [];
```

Reply:
Sure, will update. Typically I prefer to minimize the number of verbose variables hence doing it inline, will add this one explicitly if it helps for readability.

Reply:
updated in latest commit